bundle/markov/boto/firehose/firehose_client.py

```diff
@@ -0,0 +1,109 @@
+#################################################################################
+#   Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.          #
+#                                                                               #
+#   Licensed under the Apache License, Version 2.0 (the "License").             #
+#   You may not use this file except in compliance with the License.            #
+#   You may obtain a copy of the License at                                     #
+#                                                                               #
+#       http://www.apache.org/licenses/LICENSE-2.0                              #
+#                                                                               #
+#   Unless required by applicable law or agreed to in writing, software         #
+#   distributed under the License is distributed on an "AS IS" BASIS,           #
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.    #
+#   See the License for the specific language governing permissions and         #
+#   limitations under the License.                                              #
+#################################################################################
+
+'''This module implements firehose client'''
+
+import botocore
+from markov.log_handler.exception_handler import log_and_exit
+from markov.log_handler.deepracer_exceptions import GenericNonFatalException
+from markov.log_handler.constants import (SIMAPP_EVENT_SYSTEM_ERROR,
+                                         SIMAPP_EVENT_USER_ERROR,
+                                         SIMAPP_EVENT_ERROR_CODE_500,
+                                         SIMAPP_FIREHOSE_DATA_STORE_EXCEPTION)
+from markov.boto.constants import BotoClientNames
+from markov.boto.deepracer_boto_client import DeepRacerBotoClient
+
+class FirehoseClient(DeepRacerBotoClient):
```

Comment:
Does not support s3_endpoint_url

Reply:
No need for it - firehose is an AWS service, and not expect to have separate endpoint URL